packages/rolldown/src/schemas/input-options.ts

```diff
@@ -0,0 +1,273 @@
+            ]),
+          ]),
+        ),
+      ),
+    ]),
+  ),
+)
+
+const inputOptionsSchema = v.strictObject({
+  input: v.optional(inputOptionSchema),
+  plugins: v.optional(valibotExt.phantom<RolldownPluginOption>()),
+  external: v.optional(externalSchema),
+  resolve: v.optional(resolveOptionsSchema),
+  cwd: v.pipe(
+    v.optional(v.string()),
+    v.description('Current working directory'),
+  ),
+  platform: v.pipe(
+    v.optional(
+      v.union([v.literal('browser'), v.literal('neutral'), v.literal('node')]),
+    ),
+    v.description(
+      `Platform for which the code should be generated (node, ${colors.underline('browser')}, neutral)`,
+    ),
+  ),
+  shimMissingExports: v.pipe(
+    v.optional(v.boolean()),
+    v.description('Create shim variables for missing exports'),
+  ),
+  treeshake: v.optional(TreeshakingOptionsSchema),
+  logLevel: v.pipe(
+    v.optional(LogLevelOptionSchema),
+    v.description(
+      `Log level (${colors.dim('silent')}, ${colors.underline(colors.gray('info'))}, debug, ${colors.yellow('warn')})`,
+    ),
+  ),
+  onLog: v.optional(OnLogSchema),
+  onwarn: v.optional(OnwarnSchema),
+  moduleTypes: v.pipe(
+    v.optional(moduleTypesSchema),
+    v.description('Module types for customized extensions'),
+  ),
+  experimental: v.optional(
+    v.strictObject({
+      disableLiveBindings: v.optional(v.boolean()),
+      enableComposingJsPlugins: v.optional(v.boolean()),
+      resolveNewUrlToAsset: v.optional(v.boolean()),
+      strictExecutionOrder: v.optional(v.boolean()),
+    }),
+  ),
+  define: v.pipe(
+    v.optional(v.record(v.string(), v.string())),
+    v.description('Define global variables'),
+  ),
+  inject: v.optional(
+    v.record(
+      v.string(),
+      v.union([v.string(), v.tuple([v.string(), v.string()])]),
+    ),
+  ),
+  profilerNames: v.optional(v.boolean()),
+  jsx: v.optional(jsxOptionsSchema),
+  watch: v.optional(v.union([watchOptionsSchema, v.literal(false)])),
+  dropLabels: v.pipe(
+    v.optional(v.array(v.string())),
+    v.description('Remove labeled statements with these label names'),
+  ),
+
+  checks: v.optional(checksOptionsSchema),
+})
+
+const cliOverrideSchema = v.strictObject({
+  external: v.pipe(
+    v.optional(v.array(v.string())),
+    v.description(
+      'Comma-separated list of module ids to exclude from the bundle `<module-id>,...`',
+    ),
+  ),
+  inject: v.pipe(
+    v.optional(v.record(v.string(), v.string())),
+    v.description('Inject import statements on demand'),
+  ),
+  treeshake: v.pipe(
+    v.optional(v.boolean(), true),
+    v.description('enable treeshaking'),
+  ),
+})
+
+const inputCliOptionsSchema = v.omit(
+  v.strictObject({
+    ...inputOptionsSchema.entries,
+    ...cliOverrideSchema.entries,
+  }),
+  [
+    'input',
+    'plugins',
+    'onwarn',
+    'onLog',
+    'resolve',
+    'experimental',
+    'profilerNames',
+    'watch',
+  ],
+)
+
+export function validateInputCliOptions(options: any): boolean {
```

Comment:
This is only temporary method.

I'm thinking that the cli input can't actually distinguish between `inputOptions` and `outputOptions`. Should we consider combining `inputOptionsSchema` and `outputOptionsSchema` into a single file?

Reply:
Alternatively, we could use another approach by defining a custom `validateOptions(options)` method. First, we use the `inputOptionsSchema` to determine the corresponding input keys, then filter the `options` into `inputOptions` and `otherOptions`. We validate `inputOptions` and similarly extract `outputOptions` from `otherOptions` for validation. Finally, any remaining keys would be categorized as `ignoreOptions`, and we could issue a warning indicating they are being ignored.

Reply:
@hyf0 cc

Reply:
> inputOptionsSchema and outputOptionsSchema into a single file?

A giant `validator.ts ` is fine.